src/control/lib/hardware/hwloc/provider.go

```diff
@@ -42,6 +47,8 @@ func (p *Provider) GetTopology(ctx context.Context) (*hardware.Topology, error)
 	}
 }
 
+type nodeMap map[uint]*hardware.NUMANode
```

Comment:
Maybe we could make this an exported convenience type at the lib level

Reply:
Hmm, yeah, I guess so. Was going to say it's an internal implementation detail but then I realized that it's also at the core of the Topology type.

Reply:
Maybe we should be using the Topology directly?

Reply:
We could accept a Topology and build on it, like you suggested in the base PR.

Reply:
Let's do that as a follow on, in case we need to make more adjustments as we continue to integrate.